pipeline/stage03_trigger_detection/scripts/minima.py

```diff
@@ -6,17 +6,43 @@
 from utils import load_neo, write_neo, remove_annotations
 
 
-def detect_minima(asig, order):
+def detect_minima(asig, order, interpolation_points, interpolation):
     signal = asig.as_array()
+    sampling_time = asig.times[1] - asig.times[0]
+
     t_idx, channel_idx = argrelmin(signal, order=order, axis=0)
 
-    sort_idx = np.argsort(t_idx)
+    if int(interpolation) == 0:
+
+        fitted_idx_times = np.zeros([len(t_idx)])
+        
+        start_arr = np.array(t_idx) - int(interpolation_points/2)
+        start_arr = np.where(start_arr > 0, start_arr, 0)
+        stop_arr = start_arr + int(interpolation_points)
+        stop_arr = np.where(stop_arr < len(signal), stop_arr, len(signal))
+        
+        for start, stop, channel_i, i in zip(start_arr, stop_arr, channel_idx, range(len(start_arr))):
+            X = list(range(start, stop))
+            params = np.polyfit(X, signal[start:stop, channel_i], 2)
+            minimum = -(1.*params[1])/(2.*params[0])
+            fitted_idx_times[i] = minimum*sampling_time
```

Comment:
here the values are no longer indexes, let's call the variable simply `minimum_times`.
then it also get more coherent further down